<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>C_Plus_Plus_Module_Dialog</class>
 <widget class="QDialog" name="C_Plus_Plus_Module_Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>387</width>
    <height>310</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Generate CC3D C++ Module</string>
  </property>
  <widget class="QWidget" name="">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>9</y>
     <width>370</width>
     <height>293</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Module Core Name</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Module Directory</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="moduleDirLE"/>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="moduleDirPB">
        <property name="text">
         <string>Browse...</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="moduleCoreNameLE"/>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Code Layout</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QRadioButton" name="mainCodeLayoutRB">
           <property name="text">
            <string>Main Code</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="developerZoneLayoutRB">
           <property name="text">
            <string>Developer Zone</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="moduleTypeGB">
      <property name="title">
       <string>C++ Module Type</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QRadioButton" name="pluginRB">
         <property name="text">
          <string>Plugin</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="steppableRB">
         <property name="text">
          <string>Steppable</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="pythonWrapCB">
         <property name="text">
          <string>Python Wrap</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="extraAttribCB">
         <property name="text">
          <string>Attach cell attribute</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="pluginFunctionalityGB">
      <property name="title">
       <string>Plugin Functionality</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QCheckBox" name="energyFcnCB">
           <property name="text">
            <string>EnergyFunction</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="latticeMonitorCB">
           <property name="text">
            <string>LatticeMonitor</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="stepperCB">
           <property name="text">
            <string>Stepper</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="okPB">
        <property name="text">
         <string>OK</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="cancelPB">
        <property name="text">
         <string>Cancel</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <zorder>moduleTypeGB</zorder>
  <zorder>pluginFunctionalityGB</zorder>
  <zorder>groupBox</zorder>
  <zorder>extraAttribCB</zorder>
 </widget>
 <tabstops>
  <tabstop>moduleCoreNameLE</tabstop>
  <tabstop>moduleDirLE</tabstop>
  <tabstop>moduleDirPB</tabstop>
  <tabstop>pluginRB</tabstop>
  <tabstop>steppableRB</tabstop>
  <tabstop>energyFcnCB</tabstop>
  <tabstop>latticeMonitorCB</tabstop>
  <tabstop>stepperCB</tabstop>
  <tabstop>cancelPB</tabstop>
  <tabstop>okPB</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>cancelPB</sender>
   <signal>clicked()</signal>
   <receiver>C_Plus_Plus_Module_Dialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>308</x>
     <y>299</y>
    </hint>
    <hint type="destinationlabel">
     <x>163</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>steppableRB</sender>
   <signal>toggled(bool)</signal>
   <receiver>pluginFunctionalityGB</receiver>
   <slot>setHidden(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>203</x>
     <y>171</y>
    </hint>
    <hint type="destinationlabel">
     <x>69</x>
     <y>267</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
